fury/utils.py

```diff
@@ -567,3 +569,88 @@ def generate_planar_uvs(vertices, *, axis="xy"):
                 (v[2] - min_coords[2]) / range_coords[2],
             ]
     return uvs
+
+
+def create_sh_basis_matrix(vertices, l_max):
+    """Create a SH basis matrix for real spherical harmonics.
+
+    Parameters
+    ----------
+    vertices : ndarray, shape (N, 3)
+        Array of vertex coordinates in 3D space, where N is the number of vertices.
+    l_max : int
+        Maximum spherical harmonic degree.
+
+    Returns
+    -------
+    ndarray, shape (N, (l_max + 1) ** 2)
+        Matrix of spherical harmonic basis functions evaluated at the vertices.
+    """
+
+    if (
+        not isinstance(vertices, np.ndarray)
+        or vertices.ndim != 2
+        or vertices.shape[1] != 3
+    ):
+        raise ValueError("vertices must be a 2D array with shape (N, 3).")
+    if not isinstance(l_max, int) or l_max < 0:
+        raise ValueError("l_max must be a non-negative integer.")
+
+    _, theta, phi = cart2sphere(vertices[:, 0], vertices[:, 1], vertices[:, 2])
+    phi[phi < 0] += 2 * np.pi
+    n_vertices = vertices.shape[0]
+    cos_theta = np.cos(theta)
+    n_coeffs = (l_max + 1) ** 2
+    B = np.zeros((n_vertices, n_coeffs), dtype=np.float32)
+
+    col_idx = 0
+
+    for order in range(l_max + 1):
+        for m in range(-order, order + 1):
+            norm = np.sqrt(
+                ((2 * order + 1) / (4 * np.pi))
+                * (factorial(order - abs(m)) / factorial(order + abs(m)))
+            )
+
+            legendre_poly = lpmv(abs(m), order, cos_theta)
+
+            if m > 0:
+                sh_values = np.sqrt(2) * norm * legendre_poly * np.cos(m * phi)
+            elif m < 0:
+                sh_values = np.sqrt(2) * norm * legendre_poly * np.sin(abs(m) * phi)
+            else:  # m == 0
+                sh_values = norm * legendre_poly
+
+            B[:, col_idx] = sh_values
+            col_idx += 1
+
+    return B
+
+
+def get_lmax(n_coeffs, *, basis_type="standard"):
+    """Get the maximum degree (l_max) from the number of coefficients.
+
+    Parameters
+    ----------
+    n_coeffs : int
+        The number of spherical harmonic coefficients.
+    basis_type : str, optional
+        The type of spherical harmonic basis.
+        Can be "standard" or "descoteaux07". If None, defaults to "standard".
+
+    Returns
+    -------
+    int
+        The maximum spherical harmonic degree (l_max).
+    """
+
+    if not isinstance(n_coeffs, int) or n_coeffs < 1:
+        raise ValueError("n_coeffs must be a non-zero, positive integer.")
+
+    if basis_type not in ("standard", "descoteaux07"):
+        raise ValueError("basis_type must be one of 'standard' or 'descoteaux07'.")
+
+    if basis_type == "standard":
```

Comment:
what is `standard` ?

in dipy, we have also tournier. I recommend you to read this page just in case: https://docs.dipy.org/dev/theory/sh_basis.html